pymemcache/client/base.py

```diff
@@ -109,6 +109,22 @@ def _check_key(key, allow_unicode_keys, key_prefix=b''):
     return key
 
 
+def _check_integer(value, name, encoding):
+    """Check that a value is an integer and encode it as a binary string"""
+    if isinstance(value, six.integer_types):  # includes "long" on py2
+        return six.text_type(value).encode(encoding)
+    elif isinstance(value, six.text_type):
+        try:
+            int(value)
+            return six.text_type(value).encode(encoding)
```

Comment:
Because `value` is already a `six.text_type`, could we just do:

```suggestion
            return value.encode(encoding)
```

Reply:
Agh! You're right, but this makes me realize that I've introduced a pernicious bug.
I was thinking about someone passing `expire="1"`, but they could also pass `expire=b"1"` today, just as well. Under my changes `b"1"` would be rejected as invalid (but prior to this PR, it was allowed).

I doubt anyone would do _either_ of these on purpose, but I could imagine loading config or some other data and failing to properly convert it. And in that case, `b"1"` doesn't seem much less likely than `"1"`. I think this should be checking for `six.string_types` above, and still doing `six.text_type(value)` here.

Aside: this makes me think that dropping support for `str` / `bytes` / `unicode` inputs for `expire`, `delay`, and other integer params is well worth doing.

Reply:
That makes sense, and I _do_ think it would be nice to drop type conversion for these integer parameters. I see in our documentation comments that we call those fields out as `int` already.

Maybe this is something we can do for the next 2.x release given it's only been semi-supported thus far.

Reply:
I'm going to open a competing/alternate PR which enforces that int parameters are really integers. All of my fretting about "backwards compatibility" might be to the detriment of the project, long term, and I don't want to be saddling people with something subpar.